Riot/Modules/Common/Recents/DataSources/RecentsDataSource.m

```diff
@@ -1760,6 +1760,12 @@ - (void)recentsListServiceDidChangeData:(id<RecentsListServiceProtocol>)service
         NSIndexPath *indexPath = [NSIndexPath indexPathForRow:0 inSection:(NSUInteger)sectionIndex];
         [self.delegate dataSource:self didCellChange:indexPath];
     }
+    
+    if (section == RecentsListServiceSectionBreadcrumbs) {
+        [self.recentsListService cancelCurrentVoiceBroadcastRecordingIfNeededFor:self.recentsListService.breadcrumbsRoomListData];
+    } else if (section == RecentsListServiceSectionAllChats) {
+        [self.recentsListService cancelCurrentVoiceBroadcastRecordingIfNeededFor:self.recentsListService.allChatsRoomListData];
+    }
```

Comment:
Can you move this action on ListData in RecentsListService class just before calling `recentsListServiceDidChangeData` for a section?

Reply:
Done in last last commit.